packages/ajv-validator/package.json

```diff
@@ -9,10 +9,10 @@
     "dist"
   ],
   "scripts": {
-    "test": "vitest --run",
-    "build": "tsup ./src/index.ts --format esm,cjs --dts",
-    "publint": "publint",
-    "release": "yarn build && yarn test && yarn publint && yarn publish"
+    "build": "tsup ./src/index.ts",
+    "prepack": "yarn build",
+    "publint": "attw --pack && publint",
+    "test": "vitest"
```

Comment:
I'm looking to use these scripts as a template for other workspaces

- `build` - stays mostly the same. Referencing a shared config in the root, each workspace can [override config using cli arguments](https://tsup.egoist.dev/#using-custom-configuration)
- `prepack` - is run [anytime `yarn pack` is called](https://yarnpkg.com/advanced/lifecycle-scripts#prepack-and-postpack), and is triggered by both `attw` and `publint`
- `publint` - is just a convenient way to call both `attw` and `publint`. An alternative would be to have CI call the scripts individually
- `test` - isn't used by CI, it may just be handy to keep around

Reply:
What do you think of adding a Type check using the `tsc`? It now does not have some command doing `tsc` in Ajv validator middleware. But I think it's better to do a Type check, though I'm wondering if we should put it in the `test` phase or other places. If adding the `tsc`, that will fail in this Ajv validator, so we have to fix it. At the same, I think there are some projects we have to fix about Types of matter.

Reply:
I agree 👍🏻 I have plans look at using [Project References](https://www.typescriptlang.org/docs/handbook/project-references.html) to do type checking. That should allow us to use seperate tsconfig files to check individual workspaces, including any workspace dependencies. This would avoid the need to type check the entire monorepo for every build/pr/etc

I'll add it to my list 😅

Reply:
@BarryThePenguin 

Awesome! Please do so.